### PR TITLE
Crdcdh 1052

### PR DESCRIPTION
**Pull request by vshand11**
1) implemented new logics for 1052 in essential_validator.py.
2) updated check_metadata_ids fuc=nction in mongo)dao.py.

### CODE REVIEW

File: src/essential_validator.py [hunk 1]
```diff
@@ -383,16 +383,27 @@ def validate_data(self, file_info):
...
+            existed_ids = [item[NODE_ID] for item in existed_nodes]  
+            # When metadata intention is "New", all IDs must not exist in the database 
+            if len(existed_ids) > 0 and self.batch[BATCH_INTENTION] == INTENTION_NEW:
+                msg = f'“{file_info[FILE_NAME]}”: duplicated data detected: “{id_field}”: {json.dumps(duplicated_ids)}'
                 self.log.error(msg)
                 file_info[ERRORS].append(msg)
                 self.batch[ERRORS].append(msg)
                 return False
+            # When metadata intention is "Delete", all IDs must exist in the database 
+            elif self.batch[BATCH_INTENTION] == INTENTION_DELETE and len(existed_ids) < len(ids):
```
Comment (n2iw) [inline src/essential_validator.py:399]: Will it potentially fail if there are many-to-many data in the TSV? Maybe it'd be safer to remove all potential duplicated IDs before the comparison, or use sets to do comparison?

Comment (vshand11) [reply src/essential_validator.py:399]: won't.   m2m is checked in the file.   existed nodes are checked against DB.   The ids from db in the same node type and submission are unique no matter m2m relation.





[2:59](https://expanddatacommons.slack.com/archives/D05LW64MTUY/p1713812391517019)
ids from tsv file are converted to set to ensure the uniqueness.
[3:00](https://expanddatacommons.slack.com/archives/D05LW64MTUY/p1713812450308229)
not_existed_ids = list(set(ids) - set(existed_ids))

Comment (vshand11) [reply src/essential_validator.py]: updated.

**Review by n2iw — APPROVED** [review]
LGTM

**Pull request merged**